### PR TITLE
[Main Course] - 因果推断 Uplift Model

### PR DESCRIPTION
**Pull request by EchoTang04**
# Description

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.
添加Uplift model。

Fixes # (issue)

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] This change requires a documentation update

# How Has This Been Tested?

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Review by Tina0330 — APPROVED** [review]
LGTM,感谢糖糖分享

File: docs/Main_Course/Machine_Learning/2022-02-03-Uplift-Model.md [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+- One model approach
+合并实验组和对照组，将Treatment作为分类变量加入用户特征中，因此只需要对所有样本建一个模型。优点是避免误差累计，可套用现有模型。缺点是本质上依然是响应模型，而且，训练数据打通之后，很难说X和T还是不是满足条件独立假设…
+
+### 3.3 模型评估
+
+考虑到增量模型的反事实之处，我们无法通过测试集的混淆矩阵来计算准确率，召回率，AUC等等指标。但在满足CIA假定的前提下，所有样本特征同分布，我们可以对实验组和对照组样本分别预估uplift score，降序排列，分别截取十分位数（decile），计算该区间内两组样本转化率的差异，来代表这个区间的uplift，从而能对齐实验组和对照组数据，实现间接评估。对uplift绘制累计直方图，则可得到Gini曲线等来衡量不同uplift model的好坏。
+
+## 3 示例代码
+
+Python有pylift包可供使用，也有详尽的[官方文档](https://pylift.readthedocs.io/en/latest/quick-start.html)可供参考。示例代码如下。
```
Comment (jhqian0909) [inline docs/Main_Course/Machine_Learning/2022-02-03-Uplift-Model.md:70]: 最好以(https://www.kaggle.com/c/optiver-realized-volatility-prediction/discussion/302626 'Optiver Top 1 solution from nyanp') 这样reference的形式写。微信可能无法点击外链。

**Review by jhqian0909 — APPROVED** [review]

**Review by Realvincentyuan — APPROVED** [review]
谢谢Echo分享！😄

File: docs/Main_Course/Machine_Learning/2022-02-03-Uplift-Model.md (hunk 1 above)
Comment (EchoTang04) [reply docs/Main_Course/Machine_Learning/2022-02-03-Uplift-Model.md]: 已修改，谢谢佳辉提醒~

**Pull request merged**